okhttp/src/commonJvmAndroid/kotlin/okhttp3/EventListener.kt

```diff
@@ -70,6 +71,30 @@ abstract class EventListener {
   open fun callStart(call: Call) {
   }
 
+  /**
+   * Invoked for calls that were not executed immediately because resources weren't available. The
+   * call will remain in the queue until resources are available.
+   *
+   * Use [Dispatcher.maxRequests] and [Dispatcher.maxRequestsPerHost] to configure how many calls
+   * OkHttp performs concurrently.
+   */
+  open fun dispatcherQueueStart(
+    call: Call,
+    dispatcher: Dispatcher,
+  ) {
+  }
+
+  /**
+   * Invoked when [call] will be executed immediately.
+   *
+   * This method is invoked after [dispatcherQueueStart].
+   */
+  open fun dispatcherQueueEnd(
```

Comment:
It seems like this event should fire even if immediately executed. Could be useful signal for the call actually running.

And just not sent in a call.execute case

Reply:
Again, I'm stuck on the spans use case where this is triggering that something completed.